src/m365/aad/commands/policy/policy-list.ts

```diff
@@ -0,0 +1,181 @@
+      const url: string = `${this.resource}/v1.0/policies/${endpoint}`;
+      if (endpoint === "authorizationPolicy" || endpoint === "identitySecurityDefaultsEnforcementPolicy") {
+        ((): Promise<any> => {
+          return this.getPolicy(url);
+        })()
+          .then((policy: any): void => {
+            if (policy) {
+              logger.log(policy);
+            }
+            cb();
+          }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+      }
+      else {
+        ((): Promise<any[]> => {
+          return this.getPolicies(url);
+        })()
+          .then((policies: any[]): void => {
+            if (policies && policies.length > 0) {
+              logger.log(policies);
+            }
+            cb();
+          }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+      }
+    }
+    else {
+      const policyTypes: string[] = ['activityBasedTimeout', 'authorization', 'claimsMapping', 'homeRealmDiscovery', 'identitySecurityDefaultsEnforcement', 'tokenIssuance', 'tokenLifetime'];
+      let promiseCalls: any[] = [];
+      policyTypes.forEach((policyType) => {
+        let endpoint = this.getPolicyEndPoint[policyType];
+        const url: string = `${this.resource}/v1.0/policies/${endpoint}`;
+        if (endpoint === "authorizationPolicy" || endpoint === "identitySecurityDefaultsEnforcementPolicy") {
+          promiseCalls.push(
+            ((): Promise<any> => {
+              return this.getPolicy(url);
+            })()
+          );
+        }
+        else {
+          promiseCalls.push(
+            ((): Promise<any[]> => {
+              return this.getPolicies(url);
+            })()
+          );
+        }
+      });
+      Promise.all(promiseCalls)
+        .then(((results) => {
+          let allPolicies: any = [];
+          results.forEach((policies: any) => {
+            allPolicies = allPolicies.concat(policies);
+          });
+          if (allPolicies && allPolicies.length > 0) {
+            logger.log(allPolicies);
+          }
+          cb();
+        }))
+        .catch(err => {
+          this.handleRejectedPromise(err, logger, cb)
+        });
+    }
+  }
+
+  private getPolicyEndPoint: { [key: string]: string } = {
+    activityBasedTimeout: "activityBasedTimeoutPolicies",
+    authorization: "authorizationPolicy",
+    claimsMapping: "claimsMappingPolicies",
+    homeRealmDiscovery: "homeRealmDiscoveryPolicies",
+    identitySecurityDefaultsEnforcement: "identitySecurityDefaultsEnforcementPolicy",
+    tokenIssuance: "tokenIssuancePolicies",
+    tokenLifetime: "tokenLifetimePolicies"
+  }
+
+  private getPolicies(url: string): Promise<any[]> {
+    const requestOptions: any = {
+      url: url,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return new Promise<any[]>((resolve: (policies: any[]) => void, reject: (error: any) => void): void => {
```

Comment:
There is no need to wrap the request in another promise. We can simply it by doing (pseudocode, needs proper typings):

```ts
return request
  .get()
  .then(res => Promise.resolve(res.value))
```

Reply:
I have fixed this. Please review the same.